scripts/smoke-test-android.ps1

```diff
@@ -6,6 +6,11 @@ param (
     [string] $UnityVersion = ""
 )
 
+if (-not $Global:NewProjectPathCache)
```

Comment:
You could maybe move this check to `globals.ps1`, making it return early

Reply:
I don't think that works. It's not about returning early it's about the dot-scope. The whole script gets executed in the caller's scope, messing with the global variables.